src/Symfony/Component/Yaml/Inline.php

```diff
@@ -546,9 +518,8 @@ private static function parseMapping(string $mapping, int $flags, int &$i = 0, a
     private static function evaluateScalar(string $scalar, int $flags, array $references = [])
     {
         $scalar = trim($scalar);
-        $scalarLower = strtolower($scalar);
 
-        if (0 === strpos($scalar, '*')) {
+        if ('*' === $scalar[0]) {
```

Comment:
what if `$scalar` is the empty string ?

Reply:
you could use `'*' === $scalar[0] ?? null`